src/components/ha-selector/ha-selector-datetime.ts

```diff
@@ -58,8 +58,19 @@ export class HaDateTimeSelector extends LitElement {
     `;
   }
 
+  protected updated(changedProps: PropertyValues): void {
+    super.updated(changedProps);
+    if (changedProps.has("disabled") && !this.disabled && !this.value) {
```

Comment:
When enabling a disabled selector-datetime, the value in the time picker shows "12:00:00AM", but the value of the overall control remains undefined. Firing this event keeps the two in sync.